### PR TITLE
Bump seldon-deploy-sdk to 1.4.1.2 in request logger

### PR DESCRIPTION
**Pull request by adriangonz**
**What this PR does / why we need it**:

Bump the version of the SDK in the request logger to `1.4.1.2`, including a patch for auth.

**Which issue(s) this PR fixes**:

Part of #3825

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **adriangonz**
You can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by RafalSkolasinski — APPROVED** [review]
Lgtm

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/3838#pullrequestreview-848930907 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/3838#issuecomment-1009788566 "Approved")*, *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/3838#pullrequestreview-848930907 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski,axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [3838]: [Method Not Allowed]

**Comment by axsaucedo** [comment]
Issue caused due to jx reconfigured to do merge commits and only squash commits allowed in repo - allowed merge commits to fix for now 
image

**Pull request merged**
